File: src/main/java/com/ultreon/mods/screenshotmanager/client/Screenshot.java
package com.ultreon.mods.screenshotmanager.client;

import net.minecraft.client.renderer.texture.AbstractTexture;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.Nullable;

import java.io.File;

public record Screenshot(File file, @Nullable AbstractTexture texture, @Nullable ResourceLocation resourceLocation, ScreenshotData data) {

}

File: src/main/java/com/ultreon/mods/screenshotmanager/client/Screenshot.kt
package com.ultreon.mods.screenshotmanager.client

import com.badlogic.gdx.graphics.Texture
import dev.ultreon.quantum.util.NamespaceID
import java.io.File

@JvmRecord
data class Screenshot(
    val file: File,
    val texture: Texture?,
    val id: NamespaceID?,
    val data: ScreenshotData
) {
    fun dispose() {
        texture?.dispose()
    }
}